commons/src/main/java/org/infinispan/commons/util/Util.java

```diff
@@ -475,6 +476,13 @@ public static String toStr(Object o) {
          return printArray((byte[]) o, false);
       } else if (o == null) {
          return "null";
+      } else if (o.getClass().isArray()) {
```

Comment:
There's no generic `Arrays.toString(Object)`. But you're right that I can probably do the `(Object[]) o` cast, since Java arrays suffer the covariance.